crashtracker/src/collector/crash_handler.rs

```diff
@@ -414,17 +562,24 @@ fn handle_posix_signal_impl(signum: i32, sig_info: *mut siginfo_t) -> anyhow::Re
 ///     will not yet be stored.  This would lead to unexpected behaviour for the
 ///     user.  This should only matter if something crashes concurrently with
 ///     this function executing.
-pub fn register_crash_handlers(create_alt_stack: bool) -> anyhow::Result<()> {
+pub fn register_crash_handlers() -> anyhow::Result<()> {
+    static RUNNERS: AtomicU64 = AtomicU64::new(0);
+    let _ = AtomGuard::new(&RUNNERS)?; // Will return from this function if it's currently being
+                                       // called
```

Comment:
I'm confused, how would that function ever be called recursively / simultaneously? I.e. what scenario are we guarding against here?

Reply:
Ah, I was thinking about a hypothetical scenario which is unlikely.  Almost all users are going to get this as part of their top-level call into the crashtracker.  I'll remove this bit.